torch/jit/__init__.py

```diff
@@ -580,5 +567,85 @@ def _time(trace_name, name, time=True):
         print("{} {} time: {} ms".format(trace_name, name, start.elapsed_time(end)))
 
 
+def verify(model, args, loss_fn=torch.sum, devices=None):
+    """
+    Verify that a JIT compiled model has the same behavior as its uncompiled
+    version along with its backwards pass.
+
+    This function has side-effects (e.g., it executes your model / saves and loads
+    parameters), so don't expect the model to come out exactly the same as what
+    you passed in.
+
+    Arguments:
+        model (compiled torch.nn.Module): the compiled module/function to be
+            verified.  The module/function definition MUST have been decorated with
+            `@torch.jit.compile`.
+        args (tuple or Variable): the positional arguments to pass to the
+            compiled function/module to be verified.  A non-tuple is assumed to
+            be a single positional argument to be passed to the model.
+        loss_fn (function, optional): the loss function to be applied to
+            the output of the model, before backwards is invoked.  By default,
+            we assume that a model returns a single result, and we :func:`torch.sum`
+            before calling backwards; if this is inappropriate, you can pass your
+            own loss function.  Note that if a model returns a tuple of results,
+            these are passed as separate arguments to `loss_fn`.
```

Comment:
I can't see why would summing the outputs would be inappropriate. It really doesn't matter and this argument is not needed. If you really don't want to sum them, just allocate a one-filled tensors as grad_output (this is equivalent)

Reply:
This is more a question of user knobs. It would only be inappropriate if summing masked a problem with the network, which would be exposed with a different loss function. I don't know of any case where this could be true, but it seems better to let the user stick in their own loss function as needed.

Reply:
Having read your comment about using `grad`, I now understand this comment: are you saying that there should be a `grad_outputs` input to `torch.jit.verify` instead of a `loss_fn`.

Reply:
Then add an option to provide `grad_outputs`. That makes it easier to apply prepared inputs than to generate them via a loss function. We could keep both